build.rs

```diff
@@ -3,8 +3,20 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use std::env;
 use std::process::Command;
 
+fn sanity_check_for_sev_snp() {
+    let sev_snp = env::var("CARGO_FEATURE_SEV_SNP").unwrap_or("0".to_string());
+    let mshv = env::var("CARGO_FEATURE_MSHV").unwrap_or("0".to_string());
+
+    if sev_snp.eq("1") && !mshv.eq("1") {
```

Comment:
If `mshv` is required to use `sev_snp`, we should enforce this dependency from our `cargo.toml` file.

Reply:
Also, I think it is more important to check that `tdx` and `sev-snp` are not enabled at the same time (e.g. mutual exclusive). I wonder why you prefer to use the build.rs instead of using `compile_error!()` macro mentioned above (which looks cleaner to me). You can do something like:

```
#[cfg(all(feature = "tdx", feature = "sev_snp"))]
compile_error!("Feature 'tdx' and 'sev_snp' are mutual exclusive.");
```